legend-engine-test-runner-mapping/src/main/java/org/finos/legend/engine/testable/mapping/extension/MappingTestRunner.java

```diff
@@ -92,121 +86,72 @@ public TestResult executeAtomicTest(Root_meta_pure_test_AtomicTest atomicTest, P
-        if (suite instanceof MappingDataTestSuite)
-        {
-            MappingDataTestSuite mappingDataTestSuite = (MappingDataTestSuite) suite;
-            return executeMappingDataTestSuite(mappingDataTestSuite, atomicTestIds, context);
-        }
-        else if (suite instanceof MappingFunctionTestSuite)
-        {
-            MappingFunctionTestSuite mappingFunctionTestSuite = (MappingFunctionTestSuite) suite;
-            return executeMappingFuncTestSuite(mappingFunctionTestSuite, atomicTestIds, context);
-        }
-        throw new UnsupportedOperationException("Mapping Test suite type '" + suite.getClass().getName()  + "' not supported");
-    }
-
-    public List<TestResult> executeMappingDataTestSuite(MappingDataTestSuite mappingDataTestSuite, List<String> atomicTestIds, MappingTestRunnerContext context)
-    {
-        PureModel pureModel = context.getPureModel();
         List<org.finos.legend.engine.protocol.pure.v1.model.test.result.TestResult> results = Lists.mutable.empty();
-        List<Pair<Connection, List<Closeable>>> connections = buildTestConnections(mappingDataTestSuite.storeTestData, context);
-        Root_meta_pure_runtime_Runtime_Impl runtime = new Root_meta_pure_runtime_Runtime_Impl("");
-        connections.stream().forEach(conn -> runtime._connectionsAdd(conn.getOne().accept(context.connectionVisitor)));
-        List<MappingFunctionTest> functionTests = this.validateMappingSuiteTests(mappingDataTestSuite, MappingFunctionTest.class);
+        String testablePath = getElementFullPath(this.pureMapping, pureModel.getExecutionSupport());
+        Assert.assertTrue(testSuite instanceof Root_meta_pure_mapping_metamodel_MappingTestSuite, () -> "Test Suite in Mapping expected to be of type Mapping Test Suite");
+        Root_meta_pure_mapping_metamodel_MappingTestSuite compiledMappingTestSuite = (Root_meta_pure_mapping_metamodel_MappingTestSuite) testSuite;
         try
         {
-            for (MappingFunctionTest mappingFunctionTest : functionTests)
+            org.finos.legend.engine.protocol.pure.v1.model.packageableElement.mapping.Mapping mapping = ListIterate.detect(pureModelContextData.getElementsOfType(org.finos.legend.engine.protocol.pure.v1.model.packageableElement.mapping.Mapping.class), ele -> ele.getPath().equals(testablePath));
+            MappingTestSuite mappingTestSuite = ListIterate.detect(mapping.testSuites, ts -> ts.id.equals(testSuite._id()));
+            MappingTestRunnerContext context = new MappingTestRunnerContext(compiledMappingTestSuite, mapping, pureModel, pureModelContextData, extensions.flatCollect(PlanGeneratorExtension::getExtraPlanTransformers), new ConnectionFirstPassBuilder(pureModel.getContext()), extensions.flatCollect(e -> e.getExtraExtensions(pureModel)));
+            // build plan, executor args
+            List<MappingTest> mappingTests = mappingTestSuite.tests.stream().filter(t -> t instanceof MappingTest).map(t -> (MappingTest)t).collect(Collectors.toList());
```

Comment:
Wouldn't all tests be mappingTest here 
`stream().filter(t -> t instanceof MappingTest).map(t -> (MappingTest)t).collect(Collectors.toList())`
I feel this does nothing